### ISSUE TITLE
Design Review 2018-02-28 (built-in see more for amp-list, AMP template ads, amp-consent)

### ISSUE THREAD

**Issue by mrjoro** [ampproject/amphtml#13086]
Time: [2018-02-28 21:00 UTC](https://www.timeanddate.com/worldclock/meeting.html?year=2018&month=2&day=28&iv=0)
Location: [Video conference via Google Hangouts](https://hangouts.google.com/hangouts/_/google.com/amp-eng-desrev)

The AMP Project holds weekly engineering [design reviews](https://github.com/ampproject/amphtml/blob/master/contributing/design-reviews.md).  **We encourage everyone in the community to participate in these design reviews.**

If you are interested in bringing your design to design review, read the [design review documentation](https://github.com/ampproject/amphtml/blob/master/contributing/design-reviews.md) and add a link to your design doc by the Monday before your design review.

When attending a design review please read through the designs *before* the design review starts.  This allows us to spend more time on discussion of the design.

We rotate our design review between times that work better for different parts of the world as described in our [design review documentation](https://github.com/ampproject/amphtml/blob/master/contributing/design-reviews.md), but you are welcome to attend any design review.  If you cannot make any of the design reviews but have a design to discuss please let mrjoro@ know on [Slack](https://github.com/ampproject/amphtml/blob/master/CONTRIBUTING.md#discussion-channels) and we will find a time that works for you.

**Comment by lannka** [comment]
I want to talk about AMP template ads. I2I is coming. Will update here.

**Comment by zhouyx** [comment]
Talk about the new `<amp-consent>` component proposed for GDPR.

**Comment by nainar** [comment]
Time permitting I would like to talk about `<amp-bodymovin>` -> https://github.com/ampproject/amphtml/issues/12309

**Comment by cpapazian** [comment]
[Built-in see-more for amp-list](https://github.com/ampproject/amphtml/issues/13575) /cc @aghassemi

**Comment by mrjoro** [comment]
*built-in see more for amp-list*

- does it cause issues with the resize policy?
  - if end of list is still in viewport and the auto load more triggers, it might cause problems
  - there may be user experience issues with it, but the blanket AMP platform policy about resize doesn't really apply since it's post user action (this is assuming scrolling is a user action)
  - most of the times when automatic paging fires it's below the viewport, so realistically it doesn't cause a problem
- there's a brief discussion about UX, but no strong opinions on what it should be yet
- item addition/deletion on server side between two fetches (e.g. causing duplicated items); would possibly have logic on the client to de-dupe
- instead of parameters for the followup request, use a full URL?
  - yes, could do that
  - for simple cases, not having the full URL may be easier; but the full URL makes sense because there's no standard way of doing this on the server side
  - if we do a full URL bookmark then we don't need the other paging stuff (query params)
- should viewport trigger be tunable (e.g. instead of 1.5x viewport, let publisher control it--e.g. if they know their server is slower they may want to load it more quickly--but there should be a maximum
- do we want to potentially shrink the amp-list if the viewport changes, e.g. if media queries says instead of 3 column vs 4/5 columns; e.g. you create a super long list and then more columns become available, you no longer need as much vertical space
  - there are other cases we need to handle automatic sizing of amp-list, so that doesn't need to be tracked as part of this
- should there be a limit for how many things you can load?  (e.g. on low-level devices too many elements may cause issues)
  - some conversation about potentially removing things way above the viewport
  - ultimately it's the responsibility of the developer to consider this
  - can track it as a future feature enhancement
  - amp-live-list has a maxitems where things gets deleted
  - it would be better to start with a conservative limit and then relax it later (so we don't introduce a breaking change later)
  - more generally we should solve DOM recycling; Chrome team may be giving something to make this easier in the future
  - the size of the chunk affects how many you can use
  - partly depends on whether you keep the JSON around; for an actual infinite list (e.g. a photos stream) we won't address that here
- does this need to support sort/filter?
  - that's up to the page developer to handle on the server
  - client side sorting/filtering support would be nice to have

**Comment by mrjoro** [comment]
**AMP template ads**

- when an AMP page is loaded from a publisher origin, does the ad still load from the CDN?
  - friendly iframes on a publisher page?
  - will discuss more offline
  - especially with web packaging we need to consider the world where there are no cache URLs any more; this is a more general problem to think about beyond this project
- 302 redirect are painful on low end networks (serving template ads on non-AMP pages)
  - very expensive, especially if you have to do DNS resolution
  - currently in this world you're n-levels deep in redirection already, but it's still an important concern
  - @kbax will follow up with @llanka later
- ad server needs to know the request is coming from a non-AMP page and serve a different response?
  - basically just a URL parameter
  - ad networks would need to support this if they support AMP template ads
- if this takes off, the volume of requests for the templates will be large
  - templates aren't likely to change frequently
  - make it cache-control: public

**Comment by mrjoro** [comment]
**amp-consent**

- do we add "content-instance-id" to all components?
  - any that might collect user data or send requests
- is there consent callback on the component?
  - makes sense so it can wait to find out what the consent state is
- by default "if you require consent for a component and you don't have consent, don't call layout callback"
  - this is the default behavior, but then individual components could override this behavior to take it upon themselves to provide the right behavior
-  local storage
  - is there a restriction that says a single domain can access a set of keys?  how does it work when the page is on the cache vs. the publisher domain
  - don't have a very good way of syncing local storage
  - for cache domain, there's a max of 120 sites
- does GDPR only affect ads or analytics?  no, other things that collect user data will need it (iframes, other URL with substituted variables; any credentialed request to a server)
  - publisher can determine what on their page needs it
